Remove stuff

jagerman opened the pull request:
Remove stuff that we don't want or need and isn't worth supporting.

jagerman committed:
Untangle messy nested dependency tree

lokinet-cryptography depends on lokinet-base which depends on
lokinet-cryptography.

Fixed this but splitting lokinet-cryptography into the libntrup code
(which is what is really needed) and the llarp/crypto code (which isn't
needed for lokinet-base).

jagerman committed:
Remove router nickname

I don't think Kee needs it anymore.

jagerman committed:
Remove optional relay whitelist disabling

Whitelisting is now always-on for relays.  Disabling the option is never
used and is unsupported/unmaintained (it was, in theory, to allow
lokinet as a separate usage in a non-service-node relay mode, i.e. on a
completely separate network).

Confusingly, the option was enabled by the `[lokid]:enabled` config
parameter.

jagerman committed:
Remove RC key rotation

We don't have/want this anymore (and on top of that, we *never* used
it in the first place and it probably didn't even work).

dr7ana commented on llarp/router_contact.hpp:
```diff
@@ -165,18 +162,9 @@ namespace llarp
-    HasNick() const;
-
-    std::string
-    Nick() const;
```
RIP nick

dr7ana commented on llarp/router/router.cpp:
```diff
@@ -1004,22 +998,22 @@ namespace llarp
-        close_peers.emplace(pk);
-    });
+        if (conn.remote_is_relay and not _rc_lookup_handler.is_session_allowed(pk))
+          close_peers.insert(pk);
```
insert instead of emplace because it's a const reference to an already a constructed object, right?

Would using emplace copy construct a new object? Or would emplace just copy-elision it into the unordered set?

jagerman committed:
Coalesce from_string/from_string_view

The version taking a string_view will already be perfectly callable with
a string, so just combine them into one.
